pydantic_changedetect/changedetect.py

```diff
@@ -221,7 +221,8 @@ def __setattr__(self, name, value) -> None:  # noqa: ANN001
         if name in self_compat.model_fields and name not in self.model_original:
             self.model_original[name] = self.__dict__[name]
```

Comment:
This would have stored the original (unchanged) value already, even if the value did not change.

I would prefer the original not being stored at all in this case.